src/test/java/com/a2/backend/controller/UserControllerTest.java

```diff
@@ -401,4 +401,60 @@ void Test010_GivenAValidTokenAndUserWhenConfirmingUserThenStatusOKisReturned() {
         assertEquals(
                 "There is an existing user with the nickname " + anotherNickname, errorMessage);
     }
+
+    @Test
+    void GivenAvalidUserWhenWantToRecoverPasswordThenReturnStatusOk() {
```

Comment:
Overall the flow of the recovery process is wrong, it should be:
- GET recovery request to /user/recover
- At UserService generate a random recovery token and save it at the user model.
- Generated token is emailed to the user by a link to the frontend (localhost:3000/user/recover/{token})
- At the front end, user fills the form to request the password change
- POST request to /user/recover with a DTO containing: email, token and new password
- At UserService get user by email, verify the token matches and change password

Reply:
ok, so I'm not given the  DTO, the DTO is created at the userService impl?